src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -96,10 +96,14 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
             if (!this.course.learningGoals || !this.course.prerequisites || !this.course.tutorialGroups || !this.course.tutorialGroupsConfiguration) {
                 this.loadLearningGoalsAndTutorialGroups();
             }
+            await this.initAfterCourseLoad();
         } else {
-            this.loadCourse();
+            await this.loadCourse().toPromise();
+            await this.initAfterCourseLoad();
```

Comment:
Why do we need to use await here?

Reply:
For the first `await this.initAfterCourseLoad()` and the last `await this.initAfterCourseLoad()`, we don't technically need it here. However, it makes it more clear that the functions return a promise visually and it makes waiting for the whole initialization easier in the tests.

The second `await this.loadCourse().toPromise()` is needed because we need to wait for the course loading and potential redirection: Otherwise `initAfterCourseLoad` would call more endpoints that would cause 403 error alerts, which we don't want in the case of a redirect.